Shopify/ruby-lsp
# Add codeAction/resolve and variable extraction refactoring

### Motivation

Two changes are included: 
1. Implement codeAction/resolve specification that populates the "edits" field for a given code action.
2. Use the new codeAction/resolve to implement a use case of refactoring: extracting variables.

### Implementation

Limitation: Currently the use case of extracting variable does not allow customizing the new variable name, and will arbitrarily place the extracted text to the immediate line above, disregarding if it is a valid syntax.

### Automated Tests

✅

### Manual Tests
![code_action_resolve](https://user-images.githubusercontent.com/43872616/218518910-2a230eb4-e38f-46c9-8d4b-88be0a0e887a.gif)


1. Select a chunk of length in the editor. 
2. Right click to see the menu, choose `Refactor...` -> `Refactor: Extract variable`
3.  Expected: The selected text will be extracted away into a `new_variable` variable defined in the line above the selected text.
